frontend/src/pages/FormStaffPage.vue

```diff
@@ -336,18 +247,23 @@ const moveToNextStep = () => {
 };
 
 const onNext = () => {
-  //This fixes the index
-  formData.location.addresses.forEach(
-    (address: Address, index: number) => (address.index = index)
-  );
-  formData.location.contacts.forEach(
-    (contact: Contact, index: number) => (contact.index = index)
-  );
+  //This will trigger and force the validation of the current step
+  //We don't pass any field id here to force the validation of all fields
+  revalidateBus.emit();
 
-  notificationBus.emit(undefined);
-  if (currentTab.value + 1 < progressData.length) {
+  //This is now inside a setTimeout to allow the revalidateBus to finish
+  setTimeout(() => {    
+    notificationBus.emit(undefined);
+  if (currentTab.value + 1 < progressData.length) {      
     if (checkStepValidity(currentTab.value)) {
       if (reviewStatement.value) {
+        //This fixes the index
+        formData.location.addresses.forEach(
+          (address: Address, index: number) => (address.index = index)
+        );
+        formData.location.contacts.forEach(
+          (contact: Contact, index: number) => (contact.index = index)
+        );
```

Comment:
I believe this should be moved into the moveToNextStep function, so it runs whenever we actually move to the next step - instead of only when the review statement is used.